commands/application.react.js

```diff
@@ -0,0 +1,180 @@
+                project: humanizedName,
+                skipFolder: options.empty ? 'pages' : '',
+                layout: promptsResult.layout
+            });
+            modifyIndexHtml(appPath, humanizedName);
+            addTemplate(appPath, appName, templateOptions);
+        });
+    });
+};
+
+const modifyIndexHtml = (appPath, appName) => {
+    const indexHtmlPath = path.join(appPath, 'public', 'index.html');
+    let htmlContent = fs.readFileSync(indexHtmlPath).toString();
+
+    htmlContent = htmlContent.replace(/<title>(\w+\s*)+<\/title>/, `<title>${appName}<\/title>`);
+    htmlContent = htmlContent.replace('<body>', '<body class="dx-viewport">');
+    fs.writeFileSync(indexHtmlPath, htmlContent);
+};
+
+const addTemplate = (appPath, appName, templateOptions) => {
+    const templateSourcePath = path.join(__dirname, '..', 'templates', 'react');
+    const packagePath = path.join(appPath, 'package.json');
+    const manifestPath = path.join(appPath, 'public', 'manifest.json');
+    const styles = [
+        'devextreme/dist/css/dx.common.css',
+        './themes/generated/theme.additional.css',
+        './themes/generated/theme.base.css'];
+
+    moveTemplateFilesToProject(templateSourcePath, appPath, templateOptions);
+    preparePackageJsonForTemplate(packagePath, appName);
+    updateJsonPropName(manifestPath, appName);
+    install(appPath, styles);
+};
+
+const install = (appPath, styles) => {
+    appPath = appPath ? appPath : process.cwd();
+    const pathToMainComponent = path.join(appPath, 'src', 'App.js');
+    addStylesToApp(pathToMainComponent, styles);
+    addDevextremeToPackageJson(appPath);
+
+    runCommand('npm', ['install'], { cwd: appPath });
+};
+
+const getComponentPageName = (viewName) => {
+    return `${stringUtils.classify(viewName)}Page`;
+};
+
+const getNavigationData = (viewName, componentName, icon) => {
+    const pagePath = stringUtils.dasherize(viewName);
+    return {
+        route: `\n{\n    path: \'/${pagePath}\',\n    component: ${componentName}\n  }`,
+        navigation: `{\n    text: \'${stringUtils.humanize(viewName)}\',\n    path: \'/${pagePath}\',\n    icon: \'${icon}\'\n  }`
+    };
+};
+
+const createPathToPage = (pageName) => {
+    const pagesPath = path.join(process.cwd(), 'src', 'pages');
+    const newPagePath = path.join(pagesPath, pageName);
+
+    if(!fs.existsSync(pagesPath)) {
+        fs.mkdirSync(pagesPath);
+        createPagesIndex();
+    }
+
+    if(fs.existsSync(newPagePath)) {
+        console.error('The age already exists');
+        process.exit();
+    }
+
+    fs.mkdirSync(newPagePath);
+
+    return newPagePath;
+};
+
+const createPagesIndex = () => {
+    fs.writeFileSync(pathToPagesIndex, '');
+};
+
+const addView = (pageName, options) => {
+    const componentName = getComponentPageName(pageName);
+    const pathToPage = createPathToPage(pageName);
+    const pageTemplatePath = path.join(__dirname, '..', 'templates', 'pages', 'react');
```

Comment:
How about alternative folders structure? My suggestion:
`templates/react/application`
`template/react/pages`
I think it could allow making the React sources more encapsulated and not to be mixed with Vue sources in the future.